lib/modules/measure/MeasureService.ts

```diff
@@ -237,6 +224,48 @@ export class MeasureService {
               })
           );
         }
+
+        for (const [measuredAt, assetState] of Object.entries(
```

Comment:
Can you extract the logic for asset historization into a method please?

Reply:
done